src/Opulence/Framework/Databases/Bootstrappers/MigrationBootstrapper.php

```diff
@@ -58,32 +77,77 @@ public function registerBindings(IContainer $container)
     }
 
     /**
-     * Gets the executed migration repository to use in the migrator
+     * @param IContainer $container
+     */
+    protected function registerFixMigrationsCommand(IContainer $container)
+    {
+        $container->bindFactory(FixMigrationsCommand::class, function () use ($container) {
+            return new FixMigrationsCommand(
+                SqlExecutedMigrationRepository::DEFAULT_TABLE_NAME,
+                $this->getAllMigrationClasses(),
+                $container->resolve(IConnection::class),
+                $this->bindQueryBuilder($container)
+            );
+        });
+    }
+
+    /**
+     * @return array
+     */
+    protected function getAllMigrationClasses(): array
+    {
+        if (null !== $this->allMigrationClasses) {
+            return $this->allMigrationClasses;
+        }
+
+        $this->allMigrationClasses = (new FileMigrationFinder)->findAll(Config::get('paths', 'database.migrations'));
+
+        return $this->allMigrationClasses;
+    }
+
+    /**
+     * @param IContainer $container
      *
-     * @param IContainer $container The IoC container
-     * @return IExecutedMigrationRepository The executed migration repository
-     * @throws RuntimeException Thrown if there was an error resolving the query builder
+     * @return BaseQueryBuilder
      */
-    protected function getExecutedMigrationRepository(IContainer $container) : IExecutedMigrationRepository
+    protected function bindQueryBuilder(IContainer $container) : BaseQueryBuilder
     {
+        if (null !== $this->queryBuilder) {
```

Comment:
Rather than set this as an instance property, it might be best to pass it to methods that need it as a dependency.  That way, we don't have to worry about whether or not it's been set yet.